pkg/clusterinfo/clusterinfo.go

```diff
@@ -225,6 +225,7 @@ func (r *Retriever) Run(ctx context.Context) error {
 					"context cancelled, exiting cluster info trigger loop",
 					"err", ctx.Err(),
 				)
+				ticker.Stop()
```

Comment:
I think this should be a `defer ticker.Stop()` just after creation on line 220 instead. That way we can't accidentally have this same problem in the future.

Reply:
Indeed, this is not easy to occur. But our usage scenarios are a little different: We developed a collector for collecting prometheus metrics by imitating the retrieval module of prometheus and integrated some commonly used exporters, including elasticsearch_exporter, which is used as a libaray. And it is called by the main program at regular intervals, which is why such a serious memory problem occurred.

BTW, the code has been adjusted according to your suggestions. Please review the PR. Thank you.